syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -0,0 +1,135 @@
+import torch
+import syft
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+
+from typing import Union
+
+
+class FalconHelper:
+    XOR_EXPECTED_RING_SIZE = 2
+
+    @classmethod
+    def unfold(cls, image, kernel_size, padding):
+        return cls.__switch_public_private(
+            image, cls.__public_unfold, cls.__private_unfold, kernel_size, padding
+        )
+
+    @staticmethod
+    def __public_unfold(image, kernel_size, padding):
+        image = image.double()
+        image = torch.nn.functional.unfold(image, kernel_size=kernel_size, padding=padding)
+        image = image.long()
+        return image
+
+    @staticmethod
+    def __private_unfold(image, kernel_size, padding):
+        return image.unfold(kernel_size, padding)
+
+    @staticmethod
+    def xor(
+        value: Union[int, ReplicatedSharingTensor], other: Union[int, ReplicatedSharingTensor]
+    ) -> ReplicatedSharingTensor:
+        """
+        Compute the XOR value between value and other.
+        If value and other are both ints we should use the "^" operator.
+
+        Args:
+            value (ReplicatedSharingTensor or Int): RST with ring size of 2 or Int value in {0, 1}
+            other (ReplicatedSharingTensor or Int): RST with ring size of 2 or Int value in {0, 1}
+
+        Returns:
+            The XOR computation between value and other
+        """
+
+        if not any(isinstance(x, ReplicatedSharingTensor) for x in (value, other)):
+            raise ValueError("One of the arguments should be a RST")
+
+        if all((isinstance(x, ReplicatedSharingTensor) for x in (value, other))) and not all(
+            (x.ring_size == FalconHelper.XOR_EXPECTED_RING_SIZE for x in (value, other))
+        ):
+            raise ValueError("If both arguments are RST then they should be in ring size 2")
+
+        # Make sure we have a ReplicatedSharingTensor as value
```

Comment:
remove comments before merging to the codebase

Reply:
Will do